modules/module_check.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Start interactive shell to access EESSI through build container
+# mkdir -p /tmp/$USER/EESSI
+# cd /tmp/$USER/EESSI
+# git clone https://github.com/EESSI/software-layer
+# cd software-layer
+# ./eessi_container.sh
+
+# Initialize EESSI + load/configure EasyBuild
+# source /cvmfs/software.eessi.io/versions/2023.06/init/bash
+# module load EasyBuild/4.9.2
+# export WORKDIR=/tmp/$USER/EESSI
+# source configure_easybuild
+
+# .eb directory as an argument 
+if [ -z "$1" ]; then
+    echo "Usage: $0 <directory>"
+    exit 1
+fi
+
+base_dir="$1"
+
+# Dir where the modules will be 
+module_install_dir="/tmp/$USER/EESSI/module-only"
+
+locks_dir="/tmp/$USER/EESSI/locks"
+
+# Log file to record broken modules 
+broken_modules_log="broken_modules.log"
+> $broken_modules_log
+
+declare -A checked_modules
+
+# Locate all .eb files within the base dir
+easyconfig_files=$(find $base_dir -name "*.eb")
```

Comment:
This is quite a hammer depending on how this is used. It will run on the entire stack, which can take a long time as sanity checks are also run when using `--module-only`

Reply:
It would be better if we can selectively do this for easyconfigs that are introduced by a PR.

One way to do this is to gather that information using `--missing` based on the easy stack files touched by a PR.

As for implementation, to me this would have to be done as part of the build script:
* Gather the easyconfigs to be installed using `--missing` on the easystack file touched by the PR (which is already known in the build script)
* Do the builds as currently implemented 
* Go through the missing easyconfig list with `--module-only` and the other necessary options, compare the two modules and make sure that they are identical

Reply:
Specifically, I mean targeting the loop https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/EESSI-install-software.sh#L276-L295

Reply:
So add a step before 279 to gather the easyconfig files that will be built, then add an `else` clause to L283 (i.e., when the build succeeds) which rebuilds the module and checks the module contents against the original